### PR TITLE
Extend our team practices to VideoFront

### PR DESCRIPTION
**Pull request by sampaccoud**
## Purpose

Update VideoFront to FUN's Container-Native Development tool chain and code quality standards.

## Proposal

 This PR brings all the team's practices to this repository:

- [X] Add linting: gitlint, pylint (too many errors, will take time to fix), flake8, isort, black ;
- [X] CircleCI to build, lint and test the application ;
- [X] Makefile to automate build, lint, test and run for developers. 
- [X] Dockerize VideoFront ;
- [X] Add Python packaging ;
- [X] Add pytest configuration ;
- [X] Refactor settings to follow 12 factor app recommandations ;

**Note: This PR must be reviewed commit by commit**

### CODE REVIEW

File: api/models.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 from django.conf import settings
```
Comment (jmaupetit) [inline api/models.py]: `s/With/We/` in the commit message.

File: requirements/base.txt [hunk 2]
```diff
@@ -1,3 +1,6 @@
+# DEPRECATED: Python dependencies are now listed in setup.cfg. This requirement file is kept
+# only because it is still used in production. It will be removed after upgrading our
+# infrastructure to Docker.
```
Comment (jmaupetit) [inline requirements/base.txt]: :1st_place_medal:

File: docker/images/alpine/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+# ---- builder image ----
+FROM base as back-builder
+
+# Install development libraries required for compiled python dependencies (lxml,
+# Pillow and psycopg2)
+RUN apk --no-cache add --update \
+        # lxml dependencies
+        libxml2-dev \
+        libxslt-dev \
```
Comment (jmaupetit) [inline docker/images/alpine/Dockerfile]: Can you remind me which dependency requires lxml?

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,110 @@
+BOLD := \033[1m
```
Comment (jmaupetit) [inline Makefile]: In the commit message:

> [...] this Makefile can be used to install [...]

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,535 @@
+# Configuration file anchors
```
Comment (jmaupetit) [inline .circleci/config.yml]: In the commit message:

> `s/With/We/` remove

**Review by jmaupetit — APPROVED** [review]
Wow, that was a hell of a PR! I only have minor typos and remarks, else:

File: docker/images/alpine/Dockerfile (hunk 3 above)
Comment (sampaccoud) [reply docker/images/alpine/Dockerfile]: pycaption==1.0.0
  - beautifulsoup4 [required: >=4.2.1,<4.5.0, installed: 4.4.1]
  - cssutils [required: >=0.9.10, installed: 1.0.2]
  - enum34 [required: Any, installed: 1.1.6]
  - future [required: Any, installed: 0.16.0]
  - lxml [required: >=3.2.3, installed: 4.2.2]
  - six [required: >=1.9.0, installed: 1.11.0]

**Pull request merged**